### PR TITLE
docker: refactor build process to support building ARM images

### PR DESCRIPTION
**Pull request by guggero**
This PR prepares the build process to allow building ARM based docker images.
To achieve that, we need to remove the `*.proto` to JavaScript compilation step from the build process itself and require PRs that update the protos to update the code instead.

The resulting docker image(s) can be inspected here: https://hub.docker.com/layers/guggero/lightning-terminal/v0.0.0-docker-arm5/images/sha256-68baf47fc333b998c86fd07c325e4e1827ea9b0f60d9843395144ca9e7a9b701?context=explore

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -192,6 +192,14 @@ list:
 		grep -v Makefile | \
 		sort
 
+protos:
+	@$(call print, "Compiling protos.")
+	cd ./app; yarn protos
+
+protos-check: protos
```
Comment (jamaljsr) [inline Makefile:201]: Nice validation check! 👍

**Review by jamaljsr — COMMENTED** [review]
Overall, I'm in favor of this approach. It'll remove the need for users to install `protoc` to compile from source, which has been an issue for some already.  Good stuff 👍

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Instruct the GitHub linguist tool to interpret the generated JS code as just
+# that, generated. This will make sure the JS (and TS types for that matter)
+# will not show up in the repository's language statistics.
+
+app/src/types/generated* linguist-generated
```
Comment (jamaljsr) [inline .gitattributes:5]: 🙌

Thanks for finding this hidden gem.

**Review by jamaljsr — APPROVED** [review]
tACK 👍 

Tested both Dockerfile builds successfully.

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Instruct the GitHub linguist tool to interpret the generated JS code as just
```
Comment (Roasbeef) [inline .gitattributes:1]: Solid move, now I think only those doing development need to make a RAM reservation to compile `protoc` themselves...

**Review by Roasbeef — APPROVED** [review]
LGTM 🕌

**Pull request merged**